Src/DependencyCollector/Microsoft.AI.DependencyCollector/DependencyCollectorDiagnosticListener.cs

```diff
@@ -0,0 +1,252 @@
+                                SetRequestContextKeyValue(requestHeaders, RequestResponseHeaders.RequestContextSourceKey, sourceApplicationId);
+                            }
+                        }
+
+                        // Add the root ID
+                        string rootId = telemetry.Context.Operation.Id;
+                        if (!string.IsNullOrEmpty(rootId) && !requestHeaders.Contains(RequestResponseHeaders.StandardRootIdHeader))
+                        {
+                            requestHeaders.Add(RequestResponseHeaders.StandardRootIdHeader, rootId);
+                        }
+
+                        // Add the parent ID
+                        string parentId = telemetry.Id;
+                        if (!string.IsNullOrEmpty(parentId) && !requestHeaders.Contains(RequestResponseHeaders.StandardParentIdHeader))
+                        {
+                            requestHeaders.Add(RequestResponseHeaders.StandardParentIdHeader, parentId);
+                        }
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                CrossComponentCorrelationEventSource.Log.UnknownError(CrossComponentCorrelationEventSource.GetExceptionDetailString(e));
+            }
+        }
+
+        /// <summary>
+        /// Diagnostic event handler method for 'System.Net.Http.Response' event. Even in the case of an exception, this will still be called.
+        /// See https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L71 for more information.
+        /// </summary>
+        [DiagnosticName("System.Net.Http.Response")]
+        public void OnResponse(HttpResponseMessage response, Guid loggingRequestId)
+        {
+            try
+            {
+                if (response != null)
+                {
+                    DependencyTelemetry telemetry;
+                    if (this.pendingTelemetry.TryRemove(loggingRequestId, out telemetry))
+                    {
+                        string targetApplicationId = GetRequestContextKeyValue(response.Headers, RequestResponseHeaders.RequestContextTargetKey);
+                        if (!string.IsNullOrEmpty(targetApplicationId) && !string.IsNullOrEmpty(telemetry.Context.InstrumentationKey))
+                        {
+                            // We only add the cross component correlation key if the key does not represent the current component.
+                            string sourceApplicationId;
+                            if (this.correlationIdLookupHelper.TryGetXComponentCorrelationId(telemetry.Context.InstrumentationKey, out sourceApplicationId) &&
+                                targetApplicationId != sourceApplicationId)
+                            {
+                                telemetry.Type = RemoteDependencyConstants.AI;
+                                telemetry.Target += " | " + targetApplicationId;
+                            }
```

Comment:
Same here - if an exception happens trying to set cross component correlation, we can pretend as if there is no cross component correlation but still try and send the telemetry.